src/python/zigopt/handlers/experiments/create.py

```diff
@@ -177,20 +177,77 @@ def get_budget_key_and_value(cls, json_dict, runs_only):
       budget_key = get_budget_key(runs_only)
     return budget_key, budget
 
+  @classmethod
+  def _get_experiment_type(cls, experiment_type_string):
+    if experiment_type_string is None:
+      return ExperimentMeta.OFFLINE
+    try:
+      return EXPERIMENT_NAME_TO_TYPE[experiment_type_string]
+    except KeyError as ke:
+      raise BadParamError(f"Invalid experiment type: {experiment_type_string}") from ke
+
+  @classmethod
+  def _set_grid_experiment_budget(cls, budget_key, budget, experiment_meta):
+    if len(experiment_meta.conditionals) > 0:
+      raise BadParamError("Grid search experiments do not support conditional parameters currently.")
+    experiment_meta.observation_budget = GridSampler.observation_budget_from_experiment_meta(experiment_meta)
+    if budget is not None and budget != experiment_meta.observation_budget:
+      raise BadParamError(
+        f"The specified `{budget_key}` of {budget}"
+        f" does not match the computed budget of {experiment_meta.observation_budget}"
+      )
+
+  @classmethod
+  def _check_budget_required(cls, budget_key, optimized_metrics, has_constraint_metrics, num_solutions):
+    if len(optimized_metrics) > 1:
+      raise BadParamError(f"{budget_key} is required for an experiment with more than one optimized metric")
+    if has_constraint_metrics:
+      raise BadParamError(f"{budget_key} is required for an experiment with constraint metrics")
+    if num_solutions and num_solutions > 1:
+      raise BadParamError(f"{budget_key} is required for an experiment with more than one solution")
+
+  @classmethod
+  def _check_conditionals_viability(cls, experiment_meta, num_solutions):
+    """Check feature viability with conditionals"""
```

Comment:
I suppose you can remove the comments here and in 217. The function name is very close to the description